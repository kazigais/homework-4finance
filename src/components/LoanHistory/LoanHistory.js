import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button, Table } from 'semantic-ui-react';
import moment from 'moment';

class LoanHistory extends PureComponent {
  static propTypes = {
    onLoanExtend: PropTypes.func,
    loanHistory: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string,
      repayableAmount: PropTypes.number,
      amount: PropTypes.number,
      isExtended: PropTypes.bool,
      date: PropTypes.instanceOf(Date),
      dateCreated: PropTypes.instanceOf(Date),
    })),
  }

  static defaultProps = {
    onLoanExtend: () => {},
    loanHistory: [],
  }

  render() {
    const { loanHistory, onLoanExtend } = this.props;

    return (
      <Table celled striped>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell colSpan="6">Loan History</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Amount</Table.HeaderCell>
            <Table.HeaderCell>Repayable Amount</Table.HeaderCell>
            <Table.HeaderCell>Extended</Table.HeaderCell>
            <Table.HeaderCell>Payment date</Table.HeaderCell>
            <Table.HeaderCell>Date issued</Table.HeaderCell>
            <Table.HeaderCell>Options</Table.HeaderCell>
          </Table.Row>
        </Table.Header>

        <Table.Body>
          {loanHistory.map((pastLoan) => (
            <Table.Row key={pastLoan.id}>
              <Table.Cell>{pastLoan.amount}</Table.Cell>
              <Table.Cell>{pastLoan.repayableAmount}</Table.Cell>
              <Table.Cell>{pastLoan.isExtended ? 'Yes' : 'No'}</Table.Cell>
              <Table.Cell>{moment(pastLoan.date).format('LL')}</Table.Cell>
              <Table.Cell>{moment(pastLoan.dateCreated).format('LL')}</Table.Cell>
              <Table.Cell>
                <Button
                  onClick={() => (!pastLoan.isExtended ? onLoanExtend(pastLoan.id) : null)}
                  disabled={pastLoan.isExtended}
                  content="Extend"
                  primary
                />
              </Table.Cell>
            </Table.Row>
          ))}
        </Table.Body>
      </Table>
    );
  }
}

export default LoanHistory;
