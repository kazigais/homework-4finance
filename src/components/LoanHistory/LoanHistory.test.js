import React from 'react';
import LoanHistory from './LoanHistory';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('LoanHistory component', () => {

  let props;

  beforeEach(() => {
    props = {
      loanHistory: [{
        id: '123',
        repayableAmount: 100,
        amount: 80,
        isExtended: false,
        date: new Date('Feb 28 2013 19:00:00 EST'),
        dateCreated: new Date('Feb 28 2013 19:00:00 EST'),
      }],
    };
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<LoanHistory {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
