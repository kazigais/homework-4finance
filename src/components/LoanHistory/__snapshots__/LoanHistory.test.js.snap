// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanHistory component matches snapshot 1`] = `
<Table
  as="table"
  celled={true}
  striped={true}
>
  <TableHeader
    as="thead"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableHeaderCell
        as="th"
        colSpan="6"
      >
        Loan History
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableHeader
    as="thead"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableHeaderCell
        as="th"
      >
        Amount
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Repayable Amount
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Extended
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Payment date
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Date issued
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Options
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableBody
    as="tbody"
  >
    <TableRow
      as="tr"
      cellAs="td"
      key="123"
    >
      <TableCell
        as="td"
      >
        80
      </TableCell>
      <TableCell
        as="td"
      >
        100
      </TableCell>
      <TableCell
        as="td"
      >
        No
      </TableCell>
      <TableCell
        as="td"
      >
        February 28, 2013
      </TableCell>
      <TableCell
        as="td"
      >
        February 28, 2013
      </TableCell>
      <TableCell
        as="td"
      >
        <Button
          as="button"
          content="Extend"
          disabled={false}
          onClick={[Function]}
          primary={true}
        />
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;
